chef/cookbooks/magnum/recipes/post_install.rb

```
@@ -0,0 +1,67 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Cookbook Name:: magnum
+# Recipe:: post_install
+#
+
+keystone_settings = KeystoneHelper.keystone_settings(node, @cookbook_name)
+
+service_sles_image_name = "openstack-magnum-k8s-image"
+
+# Install package for magnum k8s image
+package "#{service_sles_image_name}-#{node[:kernel][:machine]}"
+
+service_sles_image_path = Dir.glob(
+  "/srv/tftpboot/files/#{service_sles_image_name}/#{service_sles_image_name}*qcow2"
+).first
+
+insecure = keystone_settings["insecure"] ? "--insecure" : ""
+
+openstack_args = "--os-username #{keystone_settings["service_user"]}"
+openstack_args += "--os-auth-type password --os-identity-api-version 3"
+openstack_args += "--os-password #{keystone_settings["service_password"]}"
+openstack_args += "--os-tenant-name #{keystone_settings["service_tenant"]}"
+openstack_args += "--os-auth-url #{keystone_settings["internal_auth_url"]}"
+openstack_args += insecure
+
+openstack_cmd = "openstack #{openstack_args}"
+
+execute "create_magnum_image" do
+  command "#{openstack_cmd} image create --file #{service_sles_image_path} --disk-format qcow2 \
+  --container-format bare --public --property os_distro=opensuse #{service_sles_image_name}"
+  not_if "#{openstack_cmd} image list --f value -c Name | grep -q #{service_sles_image_name}"
+  retries 5
+  retry_delay 10
+  action :nothing
+end
+
+execute "create_mangnum_flavor" do
+  command "#{openstack_cmd} flavor create --ram 1024 --disk 10 --vcpus 1 m1.magnum"
+  not_if "#{openstack_cmd} flavor show m1.magnum"
+  retries 5
+  retry_delay 10
```

Why retries? We should not hide errors.imho.. are we really doing that in other places already?